### ISSUE TITLE
Fluent API not working in 3.9.1

### ISSUE THREAD

**Issue by phampton24** [prisma/prisma#11641]
### Bug description

Prisma appears to have an issue resolving table names in sub selects.  Prisma is essentially telling me that I'm attempting to query for a table using the camelCase version of the name, when I should be using the PascalCase version, but I'm already using PascalCase .  

### Error Message
```
Invalid `prisma.user.findUnique()` invocation:

{
  where: {
    UserID: 8667
  },
  select: {
    userRole: {
    ~~~~~~~~
      select: {
        Role: {
          select: {
            Id: true,
            Name: true
          }
        }
      }
    },
?   UserID?: true,
?   UserName?: true,
?   FirstName?: true,
?   MiddleName?: true,
?   LastName?: true,
?   UserRole?: true,
  }
}

Unknown field `userRole` for select statement on model User. Available options are listed in green. Did you mean `UserRole`?
```

### How to reproduce

Run this query against the provided schema

```
const userRoles = await prisma.user.findUnique({
    where: { UserID: user.id }
  })
  .UserRole({
      select: {
          Role: { <shortened for brevity> }
      }
   }
});    
``` 

### Expected behavior

This query should return the requested data

### Prisma information

```
model User {
  UserID                                                                      Int                             @id(map: "PK_tblUser") @default(autoincrement())
  UserName                                                                    String?                         @unique(map: "IX_tblUser_1") @db.NVarChar(256)
  FirstName                                                                   String?                         @db.NVarChar(25)
  MiddleName                                                                  String?                         @db.NVarChar(25)
  LastName                                                                    String?                         @db.NVarChar(25)
  UserRole                                                                    UserRole[]                      @relation("UserRoleTotblUser")

  @@index([UserID, FirstName], map: "IX_tblUser_ByFirstName")
  @@index([UserID, LastName], map: "IX_tblUser_ByLastName")
  @@map("tblUser")
}

model UserRole {
  UserId         Int
  RoleId         Int
  Role           Role      @relation(fields: [RoleId], references: [Id], onUpdate: NoAction, map: "FK_UserRole_RoleId_To_Role_Id")
  tblUser        User      @relation("UserRoleTotblUser", fields: [UserId], references: [UserID], onUpdate: NoAction, map: "FK_UserRole_UserId_To_User_Id")

  @@id([UserId, RoleId], map: "PK_UserRole_UserId_RoleId")
}

model Role {
  Id              Int               @id(map: "PK_Role_Id") @default(autoincrement())
  Name            String            @unique(map: "UQ_Role_Name") @db.NVarChar(255)
  RoleEntitlement RoleEntitlement[]
  UserRole        UserRole[]
}
```

### Environment & setup

- OS:  Windows, Alpine
- Database: MS Sql Server
- Node.js version: 16.X

### Prisma Version

```
prisma                  : 3.9.1
@prisma/client          : 3.9.1
Current platform        : windows
Query Engine (Node-API) : libquery-engine bcc2ff906db47790ee902e7bbc76d7ffb1893009 (at node_modules\@prisma\engines\query_engine-windows.dll.node)
Migration Engine        : migration-engine-cli bcc2ff906db47790ee902e7bbc76d7ffb1893009 (at node_modules\@prisma\engines\migration-engine-windows.exe)
Introspection Engine    : introspection-core bcc2ff906db47790ee902e7bbc76d7ffb1893009 (at node_modules\@prisma\engines\introspection-engine-windows.exe)
Format Binary           : prisma-fmt bcc2ff906db47790ee902e7bbc76d7ffb1893009 (at node_modules\@prisma\engines\prisma-fmt-windows.exe)
Default Engines Hash    : bcc2ff906db47790ee902e7bbc76d7ffb1893009
Studio                  : 0.457.0
```

**Comment by janpio** [comment]
Are you getting an error message or unexpected data? Your issue currently only contains a schema and a query - but no further information what is going wrong besides your text description.

**Comment by phampton24** [comment]
Yes sorry, that appears to have gotten lost in all my copy and paste.  I've updated the original comment with the error message

**Comment by phampton24** [comment]
It's also worth pointing out that the above query worked as recently as version 3.8.1 of @prisma/client

**Comment by Tobbe** [comment]
I'm seeing the same thing. https://github.com/prisma/prisma/discussions/11662

**Comment by pantharshit00** [comment]
Hello,

I can reproduce this. Looks like a regression to me. I have updated the title to reflect the issue. 

Reproduction: https://github.com/harshit-test-org/prisma-issue-11641

**Comment by millsp** [comment]
Hey, the fix for this issue has been released via `3.9.2`. Thank you for reporting this issue!
